homeassistant/components/rfxtrx/config_flow.py

```diff
@@ -281,6 +286,7 @@ def _get_device_data(self, entry_id):
     def update_config_data(self, global_options=None, devices=None):
         """Update data in ConfigEntry."""
         entry_data = self._config_entry.data.copy()
+        entry_data[CONF_DEVICES] = copy.deepcopy(self._config_entry.data[CONF_DEVICES])
```

Comment:
While testing I noticed that this code didn't work at all. The data is copied, but `devices` key is not, so when updating `devices` key here it updates it also in the config entry. As a result, `async_update_entry` will never trigger a save. I needed to add this to be sure that we are editing a real copy and not the config entry.

Reply:
`.copy()` does a shallow copy. Try replacing that with with a deepcopy call.

Reply:
Nevermind, deepcopy can't pickle mappingproxy.

Reply:
See my code, it should work with proposed fix. I did a deepcopy on the `devices` key, the rest is single keys.

I tried `entry_data = copy.deepcopy(self._config_entry.data)` but it didn't work because `entry.data` is a `MappingProxy`.

Reply:
Yes, this is the correct way.